app/src/routes/project/LayersPanel/index.tsx

```diff
@@ -4,10 +4,62 @@ import { EditorMode } from '@/lib/models';
 import { PinLeftIcon, PinRightIcon } from '@radix-ui/react-icons';
 import clsx from 'clsx';
 import { observer } from 'mobx-react-lite';
-import { useState } from 'react';
+import { useCallback, useState } from 'react';
 import { useEditorEngine } from '..';
 import LayersTab from './LayersTab';
 import { capitalizeFirstLetter } from '/common/helpers';
+import { ReactComponentDescriptor } from '/electron/main/code/components';
+import { MainChannels } from '/common/constants';
+import { Button } from '@/components/ui/button';
+
+function ScanComponentsButton() {
+    const editorEngine = useEditorEngine();
+
+    const onClick = useCallback(async () => {
+        const path = (await window.api.invoke(MainChannels.PICK_COMPONENTS_DIRECTORY)) as
+            | string
+            | null;
+
+        if (path == null) {
+            return;
+        }
+
+        const components = (await window.api.invoke(
+            MainChannels.GET_COMPONENTS,
+            path,
+        )) as ReactComponentDescriptor[];
+        editorEngine.projectInfo.components = components;
+    }, [editorEngine]);
+
+    return (
+        <Button variant="outline" size="sm" className="" onClick={onClick}>
+            Connect to Project
+        </Button>
+    );
+}
+
+// TODO: make this update when `components` changes
+const ComponentsList = ({ components }: { components: ReactComponentDescriptor[] }) => {
```

Comment:
For components to subscribe to the useEditorEngine props, they need to be wrapped in observer. This also applies to children components. Each layer should subscribe as needed. 
https://github.com/onlook-dev/onlook/blob/241c7873b2e253b7d6c93bde5b03ccf696eae09b/app/src/routes/project/TopBar/index.tsx#L12

The corresponding subclass in EditorEngine also needs to instrument such that it emits changes: 
https://github.com/onlook-dev/onlook/blob/241c7873b2e253b7d6c93bde5b03ccf696eae09b/app/src/lib/editor/engine/dom/index.ts#L9

Reply:
fixed in https://github.com/onlook-dev/onlook/pull/285/commits/fee273ac862375ba4503c3ae5844f5f58e1eb009